managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1817,21 +1809,10 @@ public void operationFailed(ManagedLedgerException exception) {
 
     private void flushPendingMarkDeletes() {
         if (!pendingMarkDeleteOps.isEmpty()) {
-            if (RESET_CURSOR_IN_PROGRESS_UPDATER.get(this) == TRUE) {
-                failPendingMarkDeletes();
-            } else {
-                internalFlushPendingMarkDeletes();
-            }
+            internalFlushPendingMarkDeletes();
```

Comment:
Actually, that is covered 😉 

Take a look at `internalMarkDelete()`, there's this code there:

```java
// Trigger the final callback after having (eventually) triggered the switchin-ledger operation. This
                // will ensure that no race condition will happen between the next mark-delete and the switching
                // operation.
                if (mdEntry.callbackGroup != null) {
                    // Trigger the callback for every request in the group
                    for (PendingMarkDeleteEntry e : mdEntry.callbackGroup) {
                        e.callback.markDeleteComplete(e.ctx);
                    }
                } else {
                    // Only trigger the callback for the current request
                    mdEntry.callback.markDeleteComplete(mdEntry.ctx);
                }
```

So, 1 single write and triggers all the callbacks